### ISSUE TITLE
Clarification on licensing

### ISSUE THREAD

**Issue by lloydsargent** [wabiverse/MetaverseKit#17]
I’m really concerned about using this code for a commercial product as it depends on code that have  incompatible licenses.

I’m not sure how to build this for a commercial product that doesn’t violate one or more licenses. You have the most restrictive of licenses, GPL, with the most unrestricted, MIT.

This puts anyone who attempts to use this at legal risk without consulting with lawyers. Have you bothered to discuss this with the legal profession to determine what users legal liability is in using this code? Your license may be apache, but that doesn’t change code written using the GPL. Your license does NOT override other licenses.

This really concerns me.

**For example**

GPL 2 requires me to release SOURCE code with binaries.

> "No. The whole point of the GPL is that all modified versions must be [free software](https://www.gnu.org/philosophy/free-sw.html)—which means, in particular, that the source code of the modified version is available to the users.”

Can I provide links?

> "You're supposed to provide the source code by mail-order on a physical medium, if someone orders it. You are welcome to offer people a way to copy the corresponding source code by FTP, in addition to the mail-order option, but FTP access to the source is not sufficient to satisfy section 3 of the GPL.”

As you can see, GPL 2 is an insane license and I suggest you seriously look into it. Most companies AVOID any code that is GPL as the licenses are **incredibly** restrictive and, in the long run, expensive. The idea that a game must include source to meet this license is just absurd.

Bottom line is MIT is the best of all worlds, GPL bad—period. Other licenses can be a pain in the butt for commercial products.

**Comment by furby-tm** [comment]
Hey there! Thanks for bringing this to my attention, the purpose of this repo is to serve the existing libraries under their respective licenses as a single monolithic SwiftPM package, and is not in any way attempting to change the existing licensing of any of these libraries, hopefully that clarifies things up a bit.

Also, the Apache 2.0 license only covers the Swift code which I authored, and cannot/does not override any of the existing licenses for any of these libraries.

**Comment by furby-tm** [comment]
Additionally, I just went through and ensured the proper licensing headers are applied to all the Swift files which are authored by me, all of which are licensed under Apache 2.0, and not GPL, sorry for any confusion there.

**Comment by furby-tm** [comment]
I'm going to close this issue as I believe the following resolves any confusion here:

- https://github.com/wabiverse/MetaverseKit/commit/81508576edf0e6082af04e729b987142d83bf378

- https://github.com/wabiverse/MetaverseKit/commit/d4f1ef5ef4ed6dfdbbb8b7fc64cb201b0c356d92

These changes are apart of the latest release [**v1.8.3**](https://github.com/wabiverse/MetaverseKit/releases/tag/v1.8.3).

But if there's anything more, or more clarification you'd like on this, please don't hesitate to comment on this issue, thanks!

**Comment by lloydsargent** [comment]
I do not think you understand the problem. Under the **GPL** if you directly use **ANY** GPL file, your code **must also** be open source.

The TL;DR problem for commercial programs: it means that if I use your product (which is open source) and if it contains ANY code that is GPL, then my source code must ALSO be released.

> you MUST make the source of YOUR APPLICATION available.

My application also uses Epic’s Unreal engine code. It would be literally useless without it. My understanding is that I would also need to release Epic’s source code. Do you see the problem?

It is not _your license_ that is the problem. It is the GPL license which is why commercial house avoid any code that is GPL’ed.

[Check out this link for a better understanding what this entails.](https://softwareengineering.stackexchange.com/questions/47032/can-i-use-gpl-software-in-a-commercial-application)

**Comment by lloydsargent** [comment]
Also, just going to some of the files in your project  indicates that many license files are missing. For example, Biosc has in their headers:

_See LICENSE.txt for details about copyright and rights to use._

That file is absent. This is in itself worrisome.

**Comment by furby-tm** [comment]
@lloydsargent you keep bringing up GPL, which code _specifically_ in this repo is under GPL that you are concerned about? I’m not aware of any GPL code in this repo and you’ve now cited the GPL about 20 times.

> l Also, just going to some of the files in your project indicates that many license files are missing

Noted, thank you. I will ensure the respective license text files are included for each of the libraries in this repo.

Also I’m very well aware of what GPL is and how it works, one of my projects, the [Kraken project](https://github.com/wabiverse/Kraken) is licensed under the terms of the GPL public license.

**Comment by lloydsargent** [comment]
I am pointing this out because you really need a lawyer in order to understand why your software and the associated libraries cannot be used in a commercial environment. Good luck.

**Comment by furby-tm** [comment]
Given that you cannot even back this up with any specific examples, you really need to not harass other projects on GitHub with false information, take care.

**Comment by furby-tm** [comment]
@lloydsargent this should cover the rest of your concerns:

https://github.com/wabiverse/MetaverseKit/commit/063f69b943bb1513c360ab809e1975b37d99ebc0
